src/layout/FormFields.component.js

```diff
@@ -345,15 +351,33 @@ class FormFields extends Component {
         this.disposable.unsubscribe()
     }
 
+    handleEmailVerification = async () => {
+        await VerifyEmail()
+    }
+
     renderFields(fieldNames) {
         const d2 = this.context.d2
         const valueStore = this.props.valueStore
-
+        // Create the regular fields
         const fields = fieldNames
             .map((fieldName) => createField(fieldName, valueStore, d2))
             .filter((field) => !!field.name)
             .map((field) => wrapFieldWithLabel(field))
 
+        // Append the Verify Email button after the 'email' field
+        const emailFieldIndex = fieldNames.indexOf('email')
+        if (emailFieldIndex !== -1) {
```

Comment:
Is there a particular reason you're inserting the field with this splice operation?

I would think you should be able to add `emailVerification` after `email` in the fieldKeys in EditProfile, and then you could define a createEmailVerificationField() function that returns the component you wrote 🤔 (e.g see the definition of `createAvatarEditor` for a similar approach that also loads a functional component and passes some values from valueStore as props)

Reply:
great catch... will be considered in the next commit.